src/devices/bus/a2bus/grapplerplus.cpp

```diff
@@ -0,0 +1,236 @@
+	PORT_DIPSETTING( 0x0a, "Star Gemini" )
+	PORT_DIPSETTING( 0x02, "Anadex Printers" )
+	PORT_DIPSETTING( 0x0c, "Okidata 82A, 83A, 84, 92, 93" )
+	PORT_DIPSETTING( 0x08, "Okidata 84 w/o Step II Graphics" )
+	PORT_DIPSETTING( 0x04, "Apple Dot Matrix" )
+INPUT_PORTS_END
+
+ioport_constructor a2bus_grapplerplus_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( grapplerplus );
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - add device configuration
+//-------------------------------------------------
+
+void a2bus_grapplerplus_device::device_add_mconfig(machine_config &config)
+{
+	CENTRONICS(config, m_ctx, centronics_devices, "printer");
+	m_ctx->set_data_input_buffer(m_ctx_data_in);
+	m_ctx->busy_handler().set(FUNC(a2bus_grapplerplus_device::busy_w));
+	m_ctx->perror_handler().set(FUNC(a2bus_grapplerplus_device::perror_w));
+	m_ctx->select_handler().set(FUNC(a2bus_grapplerplus_device::select_w));
+	m_ctx->ack_handler().set(FUNC(a2bus_grapplerplus_device::ack_w));
+
+	INPUT_BUFFER(config, m_ctx_data_in);
+	OUTPUT_LATCH(config, m_ctx_data_out);
+	m_ctx->set_output_latch(*m_ctx_data_out);
+}
+
+//-------------------------------------------------
+//  rom_region - device-specific ROM region
+//-------------------------------------------------
+
+const tiny_rom_entry *a2bus_grapplerplus_device::device_rom_region() const
+{
+	return ROM_NAME( grapplerplus );
+}
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+a2bus_grapplerplus_device::a2bus_grapplerplus_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	a2bus_grapplerplus_device(mconfig, A2BUS_GRAPPLERPLUS, tag, owner, clock)
+{
+}
+
+a2bus_grapplerplus_device::a2bus_grapplerplus_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_a2bus_card_interface(mconfig, *this),
+	m_dsw1(*this, "DSW1"),
+	m_ctx(*this, GRAPPLERPLUS_CENTRONICS_TAG),
+	m_ctx_data_in(*this, "ctx_data_in"),
+	m_ctx_data_out(*this, "ctx_data_out"), m_rom(nullptr),
+	m_started(false), m_ack(0), m_irqenable(false), m_autostrobe(false), m_timer(nullptr)
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void a2bus_grapplerplus_device::device_start()
+{
+	m_rom = device().machine().root_device().memregion(this->subtag(GRAPPLERPLUS_ROM_REGION).c_str())->base();
+
+	m_timer = timer_alloc(0, nullptr);
+	m_timer->adjust(attotime::never);
+
+	save_item(NAME(m_ack));
+	save_item(NAME(m_irqenable));
+	save_item(NAME(m_autostrobe));
+}
+
+void a2bus_grapplerplus_device::device_reset()
+{
+	m_started = true;
+	m_ack = 0;
+	m_irqenable = false;
+	m_irqbit = false;
+	m_autostrobe = false;
+	lower_slot_irq();
```

Comment:
Does the device actually have a latch that gets cleared when the slot reset line is asserted, or is this a workaround for something?

Reply:
I really don't know the exact answer.  I'm just copying the a2pic device where it makes sure the irq is lowered in the reset.

src/devices/bus/a2bus/a2bus.h:	void lower_slot_irq() { m_a2bus->set_irq_line(CLEAR_LINE, m_slot); }